metrics/gen.go

```diff
@@ -0,0 +1,132 @@
+package metrics
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ipfs/go-metrics-interface"
+)
+
+var (
+	// the 1<<18+15 is to observe old file chunks that are 1<<18 + 14 in size
+	metricsBuckets = []float64{1 << 6, 1 << 10, 1 << 14, 1 << 18, 1<<18 + 15, 1 << 22}
+
+	timeMetricsBuckets = []float64{1, 10, 30, 60, 90, 120, 600}
+)
+
+// Metrics is a type which lazy initialize metrics objects.
+// It MUST not be copied.
```

Comment:
why do these need to be lazily instantiated instead of eagerly in New() ?

Reply:
in that case we can just add an option to specify the metrics obj to use, so folks could explicitly reuse them.  sounds better IMO